perun-registrar-lib/src/main/java/cz/metacentrum/perun/registrar/impl/MailManagerImpl.java

```diff
@@ -2146,14 +2241,14 @@ private String replaceValidationLinkAndRedirectUrl(Application app, ApplicationF
 					url2.append("i=").append(getUrlEncodedString(i)).append("&m=").append(getUrlEncodedString(m));
 
 				// replace validation link
-				mailText = mailText.replace(FIELD_VALIDATION_LINK, url2.toString());
+				mailText = replaceNullSafe(mailText, FIELD_VALIDATION_LINK, url2.toString());
 			}
 		}
 
 		if (mailText.contains(FIELD_REDIRECT_URL)) {
 			String redirectURL = BeansUtils.stringToMapOfAttributes(app.getFedInfo()).get("redirectURL");
 
-			mailText = mailText.replace(FIELD_REDIRECT_URL, redirectURL != null ? "&target=" + getUrlEncodedString(redirectURL) : EMPTY_STRING);
+			mailText = replaceNullSafe(mailText, FIELD_REDIRECT_URL, redirectURL != null ? "&target=" + getUrlEncodedString(redirectURL) : EMPTY_STRING);
```

Comment:
replaceNullSafe is unnecessary here I suppose

Reply:
Well, you are right. Just wanted to stay consistent in the method calls. So I believe it can be there just to be sure that if it changes in the future, `null` will not break the code again.